messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/engine/EngineWebView.kt

```diff
@@ -55,11 +70,12 @@ internal class EngineWebView @JvmOverloads constructor(
                 it.settings.domStorageEnabled = true
                 it.settings.textZoom = 100
                 it.setBackgroundColor(Color.TRANSPARENT)
-                it.addJavascriptInterface(EngineWebViewInterface(this), "appInterface")
+                engineWebViewInterface.attach(webView = it)
 
                 it.webViewClient = object : WebViewClient() {
                     override fun onPageFinished(view: WebView, url: String?) {
-                        view.loadUrl("javascript:window.parent.postMessage = function(message) {window.appInterface.postMessage(JSON.stringify(message))}")
+                        view.loadUrl("javascript:window.parent.postMessage = function(message) {window.${EngineWebViewInterface.JAVASCRIPT_INTERFACE_NAME}.postMessage(JSON.stringify(message))}")
+                        engineWebViewClientInterceptor?.onPageFinished(view, url)
```

Comment:
future refactor, we can extract the testing client from one used in code.

Reply:
It surely would make the code look much nicer, but that needs a bit more refactoring of these classes which I avoided in this PR to avoid scope creep.